pkgs/applications/science/math/giac/default.nix

```diff
@@ -1,34 +1,47 @@
 { stdenv, lib, fetchurl, fetchpatch, texlive, bison, flex, lapack, blas
 , gmp, mpfr, pari, ntl, gsl, mpfi, ecm, glpk, nauty
 , readline, gettext, libpng, libao, gfortran, perl
-, enableGUI ? false, libGL ? null, libGLU ? null, xorg ? null, fltk ? null
+, enableGUI ? false, enableMicroPy ? false, python3 ? null
+, libGL ? null, libGLU ? null, xorg ? null, fltk ? null
 }:
 
 assert enableGUI -> libGLU != null && libGL != null && xorg != null && fltk != null;
+assert enableMicroPy -> python3 != null;
```

Comment:
```suggestion
```

those packages should never be null.

Reply:
Your dislike of null is not universal, and I think that this is a bit too opinionated of a nitpick on a quite difficult PR. Can you let the author concentrate on the actual issues at stake please?

Reply:
I was just following the existing package convention of not depending on packages that are only conditionally needed. I can definitely see the argument that Python 3 is ubiquitous while "xorg" might not be available on certain platforms, though. I will let @symphorien decide though, since he's this package's maintainer.

Reply:
> Your dislike of null is not universal

``? null`` for optional features it is not required and even discouraged because hides errors when renaming and removing packages and breaks overrides in certain cases. There are certain use cases where it is required but I am pretty sure in this case it will not cause any regressions and make sage easier to maintain.

> I was just following the existing package convention of not depending on packages that are only conditionally needed.

nixpkgs goes the other way most of the time and just includes most optional features by default.

> "xorg" might not be available on certain platforms

even if it is, it will never be null and marked broken or platforms will not allow it.

If an input package is null this is a bug and should be investigated.

Reply:
Fair enough. I don't have a horse in this race, so I will let the giac package maintainer decide if removing `? null` is a desirable change. (By the way, is there a treewide effort to remove `? null` in such cases? I ask because knowing of such an effort would make the decision trivial.)

@symphorien What do you think? I don't mind at all doing an extra rebase. @SuperSandro2000 already changed it in #115361, which I think is an overstepping, but I urge you to consider this merely on technical grounds.

My opinion (which I am **not** claiming matters) is that users should not need to download `fltk` or `xorg.libX11` if they want to use `giac` and not `giac-with-xcas`, and if removing `? null` does keep this property (and does not break Darwin) I don't really mind.

Reply:
My point is not on the technical merits of adding or removing `null`, just that this PR has enough meat that such debatable nitpicking might be detrimental. But I guess my comment was counterproductive, so let's cut it short: do as you wish.

Reply:
@symphorien I appreciate you stepping in to make sure the PR wasn't derailed by the `null` point, thank you! I offered to do an extra rebase for the same reason, in fact.